module_report/check_apis.py

```diff
@@ -0,0 +1,15 @@
+import os
+import subprocess
+
+
+def check_apis(report_name):
+    apis = [
+        'pyqt5',
+        'pyqt6',
+        'pyside2',
+        'pyside6',
+    ]
+
+    for api in apis:
+        os.environ['QT_API'] = api
+        subprocess.call(f"python module_report/check_api.py {report_name}", shell=True)
```

Comment:
@DaelonSuzuka 

> I originally tried to write this section without launching a new interpreter at all

Yeah, I figured this would be necessarily due to the complexity and sensitivity of the QtPy startup and Qt import process, so I didn't question your decision here.

> `sys.executable` is new to me, thanks.

I've had to interact with it all the time, to help users debug issues and set interpreter paths, as well as programmatically for various things with Spyder and elsewhere, so I guess I just assumed that an experianced Python like yourself would be aware of it, sorry.

> Do you have any recommended reading on launching python from python? I'm sure this isn't the last time I'll need to do this.

For launching a brand new interpreter with no connection to the previous, aside from possibly env variables and CLI args. it just boils down to `subprcoess.run([sys.executable, *python_args_here], **subprocess_kwargs_here)`. That's how we do it in Spyder for, e.g. LSP and environment detection. Per the [admonition in the subprocess docs](https://docs.python.org/3/library/subprocess.html#subprocess.Popen),

> On all platforms, passing [sys.executable](https://docs.python.org/3/library/sys.html#sys.executable) is the recommended way to launch the current Python interpreter again, and use the -m command-line format to launch an installed module.

If you want/need something more tightly coupled, where the processes can exchange data (IPC) or even share state, there's too many options to describe—threading, multiprocessing, concurrent.futures, asyncio, third party async frameworks, subinterpreters, third-party IPC systems like ZMQ (as we use with Spyder/Spyder-Kernels) and RabbitMQ, OS-level mechanisms like pipes and green threads, etc, etc...but for just the case of launching a brand new, independent Python interpreter, it basically just boils down to the line above, with whatever CLI args, env vars and other setup you want.

Reply:
Oh, `sys.executable` is just the path of the current interpreter. Clean.

I don't think I've ever needed to run python from inside python until now, it's always been python -> something else or vice versa.